src/systemctl/systemctl.c

```diff
@@ -1001,18 +1001,24 @@ static int systemctl_parse_argv(int argc, char *argv[]) {
 
                 case ARG_WHEN:
                         if (streq(optarg, "show")) {
-                                r = logind_show_shutdown();
-                                if (r < 0 && r != -ENODATA)
-                                        return r;
-
-                                return 0;
+                                arg_action = ACTION_SYSTEMCTL_SHOW_SHUTDOWN;
+                                return 1;
                         }
 
                         if (STR_IN_SET(optarg, "", "cancel")) {
-                                arg_when = USEC_INFINITY;
+                                arg_action = ACTION_CANCEL_SHUTDOWN;
+                                return 1;
+                        }
+
+                        if (streq(optarg, "auto")) {
+                                if (logind_has_maintenance_window())
```

Comment:
please don't check ahead of time. instead, just try to schedule the reboot assuming that there is a maintainance window. Which might then fail with NoDesignatedMaintainanceTimeSet, in which case you try with the 1min thing.